### PR TITLE
Add Workspace Members Search Feature

### PR DESCRIPTION
**Pull request by PankajAS**
### Fixed Issues
$ https://github.com/Expensify/App/issues/12339
PROPOSAL: [#12339(comment)](https://github.com/Expensify/App/issues/12339#issuecomment-1422805292)

### Tests
1. Navigate to Settings.
2. Click on Workspaces.
3. Click on any Workspace and Manage members.
4. Search member by name or email address.

- [] Verify that no errors appear in the JS console

### Offline tests
1. Navigate to Settings.
2. Click on Workspaces.
3. Click on any Workspace and Manage members.
4. Search member by name or email address.

### QA Steps
1. Navigate to Settings.
2. Click on Workspaces.
3. Click on any Workspace and Manage members.
4. Search member by name or email address.

- [] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/17126628/218137774-ba89c425-880c-4c65-a526-b91aa097b2aa.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/17126628/218143163-82d0d273-dfb8-488d-972a-3d6d57eac7b1.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/17126628/218138191-8e2704d8-61ab-4cec-abdd-4bf43b4f5bc4.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/17126628/218138700-628d7aca-7f51-4c78-a849-0e940c521af6.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/17126628/218139516-f704b589-0fbb-4288-965e-d8c02c791062.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/17126628/218138792-fc8b5061-99bc-462a-80a4-305ae6d91663.mov

</details>

### CODE REVIEW

**Comment by stitesExpensify** [comment]
Hi @PankajAS what issue is this from? The linked URL is invalid

**Comment by PankajAS** [comment]
@stitesExpensify its from issue [#12339](https://github.com/Expensify/App/issues/12339) and here is  [proposal](https://github.com/Expensify/App/issues/12339#issuecomment-1422805292)

**Comment by PankajAS** [comment]
Hi @stitesExpensify , have you get chance to review PR?

**Review by thesahindia — CHANGES_REQUESTED** [review]
We also need to add a header message. "No results found" should appear when there is no result for a search term.

**Comment by PankajAS** [comment]
> I think we also need to do something for the "Select all" button because it looks kinda weird when there are no results.
> Screen.Recording.2023-02-13.at.9.06.07.PM.mov
> 
> We need help with the design and expected behaviour
> 

> cc: @shawnborton

@thesahindia  We can hide "Select All" option when there is no record found?
Screenshot 2023-02-13 at 9 50 50 PM

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 1]
```diff
@@ -299,6 +307,23 @@ class WorkspaceMembersPage extends React.Component {
             });
         });
         data = _.sortBy(data, value => value.displayName.toLowerCase());
+        data = _.filter(data, (member) => {
```
[the remark this answers is not in the document]
Comment (PankajAS) [reply src/pages/workspace/WorkspaceMembersPage.js:310]: if we use `getParticipantNames` in Workspace member then we have to again filter `data` as its using with `Flatlist`, that's why i used direct filter with `data`

**Comment by PankajAS** [comment]
@thesahindia i have added "No results found" header and hide "Select All" when there is no record found, please check

**Comment by PankajAS** [comment]
> I will be OOO for 1-2 days. @0xmiroslav will be reviewing this.

Ok @thesahindia , 

Hi @0xmiroslav can you please review PR

**Comment by shawnborton** [comment]
Can someone share the latest screenshots with me and I can provide design feedback from there? Thanks!

**Comment by PankajAS** [comment]
@shawnborton here is latest UI video

https://user-images.githubusercontent.com/17126628/218642449-f60e82c3-5331-49f4-b07a-f97b96b5f10a.mov

**Comment by PankajAS** [comment]
> @PankajAS please pull from `main` and fix conflicts. And share latest screenshots for @shawnborton

@0xmiroslav i have fixed conflicts, check now

**Comment by PankajAS** [comment]
> @PankajAS It seems PR is broken. I think you're doing merge in wrong way. Can you close this one and create new PR based on latest `main`? merge bug

@0xmiroslav I just rebase with `main` branch and fix conflicts and push code in this branch, anyway let me create new pull request.

**Comment by PankajAS** [comment]
> @PankajAS Don't use `rebase`

Ok @0xmiroslav, let me create new pull request if that's ok?

**Pull request closed without merging**